hazelcast/src/main/java/com/hazelcast/nio/serialization/ClassDefinitionBuilder.java

```diff
@@ -213,7 +341,10 @@ public ClassDefinition build() {
         return cd;
     }
 
-    private void check() {
+    private void check(String fieldName) {
+        if (!addedFieldNames.add(fieldName)) {
+            throw new HazelcastSerializationException("Field with field name : " + fieldName + " already exists");
```

Comment:
`HazelcastSerializationException` isn't correct IMO. It should be thrown "when a problem occurs while serializing/deserializing objects". This should rather be `IllegalStateException`. 

But it's consistent with the same issue 3 lines below...